relay/src/lib.rs

```diff
@@ -152,89 +135,6 @@ pub async fn resolve_image_output(input: &str, guest_entry: &GuestListEntry) ->
         "bonsai" => tokio::task::spawn_blocking(move || prove_alpha(elf, input))
             .await
             .expect("Failed to run alpha sub-task"),
-        "local" => prove_locally(elf, input, true),
-        _ => prove_locally(elf, input, false),
-    }
-}
-
-abigen!(ProxyContract, "artifacts/proxy.sol/Proxy.json");
-
-pub struct Config {
-    pub proxy_address: Address,
-}
-
-pub async fn run_with_ethers_client<M: Middleware + 'static>(config: Config, ethers_client: Arc<M>)
-where
-    <M as ethers::providers::Middleware>::Provider: PubsubClient,
-    <<M as ethers::providers::Middleware>::Provider as ethers::providers::PubsubClient>::NotificationStream: Sync,
-{
-    let event_name = "CallbackRequest(address,bytes32,bytes,address,bytes4,uint64)";
-    let filter = ethers::types::Filter::new()
-        .address(config.proxy_address)
-        .event(event_name);
-    let mut proxy_stream = ethers_client
-        .subscribe_logs(&filter)
-        .await
-        .expect("Failed to subscribe to ethereum event logs")
-        .map(|log| {
-            ethers::contract::parse_log::<CallbackRequestFilter>(log)
-                .expect("must be a callback proof request log")
-        });
-
-    let proxy: ProxyContract<M> = ProxyContract::new(config.proxy_address, ethers_client.clone());
-    while let Some(event) = proxy_stream.next().await {
-        // Search list for requested binary name
-        let image_id = hex::encode(event.image_id);
-        let guest_entry =
-            resolve_guest_entry(GUEST_LIST, &image_id).expect("Failed to resolve guest entry");
-
-        // Execute or return image id
-        let input = hex::encode(event.input);
-        let journal_bytes = resolve_image_output(&input, guest_entry)
-            .await
-            .expect("Failed to compute journal output");
-
-        let payload = [
-            event.function_selector.as_slice(),
-            journal_bytes.as_slice(),
-            event.image_id.as_slice(),
-        ]
-        .concat();
-
-        // Broadcast callback transaction
-        let proof_batch: Vec<Callback> = vec![Callback {
-            callback_contract: event.callback_contract,
-            journal_inclusion_proof: vec![],
-            payload: payload.into(),
-            gas_limit: event.gas_limit,
-        }];
-
-        proxy
-            .invoke_callbacks(proof_batch)
-            .send()
-            .await
-            .expect("failed to send callback transaction")
-            .await
-            .expect("Failed to confirm callback transaction");
+        _ => execute_locally(elf, input),
```

Comment:
Default case should be an error so that if someone makes a typo, it fails in an obvious way (e.g. `BONSAI_PROVING=bosnai` should quickly error) 
```suggestion
        "local" | "" => execute_locally(elf, input),
        _ => bail!("valid options for BONSAI_PROVING are 'bonsai' and 'local', got: {}", prover.as_str()),
```